packages/flutter/lib/src/material/floating_action_button.dart

```diff
@@ -319,10 +314,40 @@ class FloatingActionButton extends StatelessWidget {
 
   final BoxConstraints _sizeConstraints;
 
+  static const double _defaultElevation = 6;
+  static const double _defaultHighlightElevation = 12;
+  static const ShapeBorder _defaultShape = CircleBorder();
+  static const ShapeBorder _defaultExtendedShape = StadiumBorder();
+
   @override
   Widget build(BuildContext context) {
     final ThemeData theme = Theme.of(context);
-    final Color foregroundColor = this.foregroundColor ?? theme.accentIconTheme.color;
+    final FloatingActionButtonThemeData floatingActionButtonTheme = theme.floatingActionButtonTheme;
+
+    final Color backgroundColor = this.backgroundColor
+      ?? floatingActionButtonTheme.backgroundColor
+      ?? theme.colorScheme.secondary;
+    final Color foregroundColor = this.foregroundColor
+      ?? floatingActionButtonTheme.foregroundColor
+      ?? theme.accentIconTheme.color
+      ?? theme.colorScheme.onSecondary;
+    final double elevation = this.elevation
+      ?? floatingActionButtonTheme.elevation
+      ?? _defaultElevation;
+    final double highlighElevation = this.highlightElevation
+      ?? floatingActionButtonTheme.highlightElevation
+      ?? _defaultHighlightElevation;
+    final double disabledElevation = this.disabledElevation ?? elevation;
+    final MaterialTapTargetSize materialTapTargetSize = this.materialTapTargetSize
+      ?? theme.materialTapTargetSize;
+    final TextStyle textStyle = theme.accentTextTheme.button.copyWith(
```

Comment:
Is there no way to customize the textStyle on the widget?

Reply:
not currently

Reply:
Could you add a TODO for that?

Reply:
I could add a GH issue

Reply:
#29371